cosmo/filesystem.py

```diff
@@ -63,35 +68,45 @@ def __init__(
             self, filename, hdu, keys, exts, spt_keys=None, spt_exts=None, data_ext=None, data_keys=None
     ):
         self.filename = filename
+        self.spt_file = self._define_spt_filename()
         self.hdu = hdu
         self.header_keys = keys
         self.header_exts = exts
         self.spt_keys = spt_keys
         self.spt_exts = spt_exts
         self.data_keys = data_keys
         self.data_exts = data_ext
-        self.data = dict()
+        self.data = {}
 
-    def get_spt_header_data(self):
+    def _define_spt_filename(self):
         path, name = os.path.split(self.filename)
         spt_name = '_'.join([name.split('_')[0], 'spt.fits.gz'])
         spt_file = os.path.join(path, spt_name)
 
         if not os.path.exists(spt_file):
             return
 
-        with fits.open(spt_file) as spt:
+        return spt_file
+
+    def get_spt_header_data(self):
+        with fits.open(self.spt_file) as spt:
             self.data.update({key: spt[ext].header[key] for key, ext in zip(self.spt_keys, self.spt_exts)})
 
     def get_header_data(self):
         self.data.update({key: self.hdu[ext].header[key] for key, ext in zip(self.header_keys, self.header_exts)})
 
-    def get_file_data(self):
+    def get_table_data(self):
         self.data.update({key: self.hdu[ext].data[key] for key, ext in zip(self.data_keys, self.data_exts)})
 
 
 @dask.delayed
-def get_file_data(fitsfile, keys, exts, exp_type=None, spt_keys=None, spt_exts=None, data_ext=None, data_keys=None):
+def get_file_data(fitsfile: str, keys: Sequence, exts: Sequence,
+                  exp_type: Sequence = None,
+                  spt_keys: Sequence = None,
+                  spt_exts: Sequence = None,
+                  data_ext: Sequence = None,
+                  data_keys: Sequence = None) -> Union[dict, None]:
+    """Get specified data from a fitsfile and optionally its corresponding spt file."""
     with fits.open(fitsfile, memmap=False) as file:
         if exp_type and file[0].header['EXPTYPE'] != exp_type:
             return
```

Comment:
Nope. We don't want it to break here. We want it to continue on it's merry way and do nothing.

However, the way that I'm thinking about it could be wrong, so we should talk about it

Reply:
I'm too lazy to expand the code and find the context, it might be the case that this should throw and something else catches it, and then that decides to go on its merry way.